pandas/core/series.py

```diff
@@ -887,21 +887,20 @@ def __getitem__(self, key):
         elif key_is_scalar:
             return self._get_value(key)
 
-        if (
-            isinstance(key, tuple)
-            and is_hashable(key)
-            and isinstance(self.index, MultiIndex)
-        ):
+        if not is_iterator(key) and is_hashable(key):
             # Otherwise index.get_value will raise InvalidIndexError
-            try:
-                result = self._get_value(key)
-
-                return result
+            if isinstance(self.index, MultiIndex):
```

Comment:
instead of doing the if first, can you do it in the keyerror itself e.g.
```
try:
    result = self._get_value(key)
    return result
except KeyError:
    if isinstance(self.index, MultiIndex):
        return self._get_values_tuple(key)
```

of course maybe we should actually do this *inside* _get_value

Reply:
Since this is called from 4 places, I'll look into doing this for 1.2 in a followup.